From e3904cd963188dcbfcfa99cca0b2e030818b7bb2 Mon Sep 17 00:00:00 2001
From: Tal Yacobi <tyacobi@habana.ai>
Date: Tue, 13 May 2025 11:22:17 +0300
Subject: [PATCH] [SW-221571] gaudi3: use is ina present function

Use the refactored is ina236 present function
instead of directly checking if the card type is pcie.

Change-Id: I2148150ac82f033076e8a43adfd3584c9a863c1d
Signed-off-by: Tal Yacobi <tyacobi@habana.ai>
---

diff --git a/arch/gaudi3/sensors.c b/arch/gaudi3/sensors.c
index 2d18a69..39d8ce3 100644
--- a/arch/gaudi3/sensors.c
+++ b/arch/gaudi3/sensors.c
@@ -157,7 +157,7 @@
 
 static bool hl_ina236_valid(uint32_t idx)
 {
-	return idx == 0 ? !hl_is_card_type_pcie() : true;
+	return idx == 0 ? hl_is_ina236_0_present() : true;
 }
 
 static void hl_volt_asc10_compute(struct sensor_value *value, uint16_t sensor_idx)
